lib/meli-client.js

```diff
@@ -0,0 +1,111 @@
+const Promise = require('bluebird')
+const req = require('request-promise')
+const Account = require('../model/account')
+const {refresh} = require('../config/meli-auth')
+
+const GET_ORDERS_URL = `https://api.mercadolibre.com/orders/search`
+
+class MeliClient {
+  constructor() {
+    this.accounts = []
+  }
+
+  /**
+   *
+   * @param filters
+   * @returns {Promise<void>}
+   */
+  async getOrders({accounts = [], id} = {}) {
+    await this._updateAccounts()
+    const filterAccountsIds = accounts.map(a => a.id)
+    const filteredAccounts = filterAccountsIds.length > 0 ?
+      this.accounts.filter(acc => filterAccountsIds.includes(acc.id)) :
+      this.accounts
```

Comment:
Make the `filterAccountsIds.length > 0` validation before the map, filter, includes, etc.

Reply:
it's being done already (line 21)

Reply:
You are making an unnecessary accounts.map at line 20 in some cases, just because you're not making the validation before

Reply:
The map in line 20 is not done unnecesarily, if the list is empty then an empty list will be returned..